usd/src/sdf_parser/Material.cc

```diff
@@ -0,0 +1,462 @@
+  /// \param[in] _customData Custom data to set the field
+  /// \param[in] _displayName Display name
+  /// \param[in] _displayGroup Display group
+  /// \param[in] _doc Documentation of the field
+  /// \param[in] _colorSpace if the material is a texture, we can specify the
+  /// color space of the image
+  template<typename T>
+  void CreateMaterialInput(
+    const pxr::UsdPrim &_prim,
+    const std::string &_name,
+    const pxr::SdfValueTypeName &_vType,
+    const T &_value,
+    const std::map<pxr::TfToken, pxr::VtValue> &_customData,
+    const pxr::TfToken &_displayName,
+    const pxr::TfToken &_displayGroup,
+    const std::string &_doc,
+    const pxr::TfToken &_colorSpace = pxr::TfToken(""))
+  {
+    auto shader = pxr::UsdShadeShader(_prim);
+    if (!shader)
+    {
+      std::cerr << "Not able to convert the prim to a UsdShadeShader\n";
+    }
+
+    auto existingInput = shader.GetInput(pxr::TfToken(_name));
+    pxr::SdfValueTypeName vTypeName;
+    if (_vType.IsScalar())
+    {
+      vTypeName = _vType.GetScalarType();
+    }
+    else if (_vType.IsArray())
+    {
+      vTypeName = _vType.GetArrayType();
+    }
+    auto surfaceInput = shader.CreateInput(
+      pxr::TfToken(_name), vTypeName);
+    surfaceInput.Set(_value);
+    auto attr = surfaceInput.GetAttr();
+
+    for (const auto &[key, customValue] : _customData)
+    {
+      attr.SetCustomDataByKey(key, customValue);
+    }
+    if (!_displayName.GetString().empty())
+    {
+      attr.SetDisplayName(_displayName);
+    }
+    if (!_displayGroup.GetString().empty())
+    {
+      attr.SetDisplayGroup(_displayGroup);
+    }
+    if (!_doc.empty())
+    {
+      attr.SetDocumentation(_doc);
+    }
+    if (!_colorSpace.GetString().empty())
+    {
+      attr.SetColorSpace(_colorSpace);
+    }
+  }
+
+  pxr::UsdShadeMaterial ParseSdfMaterial(
+    const sdf::Material *_material, pxr::UsdStageRefPtr &_stage)
+  {
+    auto looksPrim = _stage->GetPrimAtPath(pxr::SdfPath("/Looks"));
+    if (!looksPrim)
+    {
+      looksPrim = _stage->DefinePrim(
+        pxr::SdfPath("/Looks"), pxr::TfToken("Scope"));
+    }
+
+    // This variable will increase with every new material to avoid collision
+    // with the names of the materials
+    static int i = 0;
+
+    const std::string mtl_path = "/Looks/Material_" + std::to_string(i);
+
+    auto usdMaterialPrim = _stage->GetPrimAtPath(pxr::SdfPath(mtl_path));
+    pxr::UsdShadeMaterial material;
+    if (!usdMaterialPrim)
+    {
+      material = pxr::UsdShadeMaterial::Define(_stage, pxr::SdfPath(mtl_path));
+    }
+    else
+    {
+      material = pxr::UsdShadeMaterial(usdMaterialPrim);
+    }
+
+    auto usdShader = pxr::UsdShadeShader::Define(
+      _stage,
+      pxr::SdfPath(mtl_path + "/Shader"));
+    auto shaderPrim = _stage->GetPrimAtPath(pxr::SdfPath(mtl_path + "/Shader"));
+    if (!shaderPrim)
+    {
+      std::cerr << "Not able to cast the UsdShadeShader to a Prim" << '\n';
```

Comment:
Same comment applies here about changing `sdf::cerr` to `UsdError` (see https://github.com/ignitionrobotics/sdformat/pull/831#discussion_r816438465)